### DISCUSSION TITLE
Forks of AltDrag

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/altdrag#128] [Show and tell]
Hey everyone.

In the spirit of #127, I want to have a discussion thread about forks of AltDrag. If you have forked AltDrag then feel free to make a post here to advertise your fork. Describe the changes you've made and why people should use it.

Let's try to reserve the top posts for forks. Feel free to discuss specific forks by commenting directly on that post. I may delete top level comments that are off topic.

**If you fork AltDrag, please use the Fork button on GitHub. That makes it a lot easier to track and perhaps eventually reintegrate some of your changes to the official release.**

Thanks!

**Comment by RamonUnch** [comment]
I made **AltSnap**, a fork of **AltDrag**, available here:
https://github.com/RamonUnch/AltSnap/
I did not use the *fork* option because it started by a big refactoring that was done offline.
There are some new actions and it fixes most of Win8/10 problems:
* Fixed Win10 Invisible borders
* Fixed High DPI support
* Fixed freezes in Win8/10.
* Maximized and snapped Windows now move only when you drag them instead of a single Alt+Click
* Fixed Lower with MMB action for some programs such as SumatraPDF
* Fixed: now there is a max speed beyond which snapping no longer occurs.
* Fixed: No longer snap to Cloaked windows (Window 8.x and later)
+ Added Smart Aero sizing (snap to other resized snapped windows)
+ Added Sticky Aero sizing
+ Added transparent window while moving.
+ Added activation on Key combo
+ Added activation on mouse click (hotclick)
+ Added Maximize, Roll, Borderless, Menu and Kill actions
+ Added Killkeys: list of keys that unhook AltDrag
+ Added the option to change some metrics
+ Added Snap Layout and FancyZones integration
- Removed the HooksWindow option
- Removed focus on typing option
- Removed Auto Update stuff (use the Chocolatey package for this: https://community.chocolatey.org/packages/altdrag/).

EDIT October 6 2021: I renamed the fork from AltDrag to **AltSnap** and added the Snap Layout option

**Reply by stefansundin** [reply]
Hi. It seems like your fork is the most popular one so far, which is great.

I just have one comment though; I wish the chocolatey package had been named something else. I guess I can blame myself for not trying to take the name earlier. Maybe it's ok for right now since I am not actively improving the official AltDrag release. But could you update the chocolatey description to make it extra clear that it is not the official AltDrag?

Thank you!

**Reply by RamonUnch** [reply]
I could give you the ownership of the package, I did not think about the name however, I do have quite some users that use this package and changing the name would bug them.
I clarified the descriptionfor the [1.42 release](https://community.chocolatey.org/packages/altdrag/1.42).

**Reply by stefansundin** [reply]
It's ok for now. Maybe in the future if I release new versions. Thank you for updating the description.

**Reply by RamonUnch** [reply]
I just made a PR to the main AltDrag repo to fix the huge freezing bug and to have the proper .manifest file.
I understand that my fork went quite some distance from the original, so If I see more trivial fixes like this, I will create the PRs.

**Reply by bigblondewolfpersonal** [reply]
@RamonUnch thanks a lot for your work! I am having serious trouble with AltDrag on my corporate laptop, which features the worst combination: frequent display changes (home vs ofice); 4k monitor and OpenText Exceed TurboX remote connection to a Linux computer that doesn't play nicely with addons. AltDrag frequently hangs and causes my keyboard to freeze until I kill it. I was planning to contact Stefan and do some debug, but saw your fork and AltSnap is flawless for the past days.
Great job both!
